enforcer/datapath_tcp.go

```diff
@@ -231,6 +232,25 @@ func (d *Datapath) processApplicationTCPPacket(tcpPacket *packet.Packet, context
 // processApplicationSynPacket processes a single Syn Packet
 func (d *Datapath) processApplicationSynPacket(tcpPacket *packet.Packet, context *PUContext, conn *TCPConnection) (interface{}, error) {
 
+	// Let's check if it matches a specific external service - we can let those go
+	// We don't account for 0.0.0.0/0 external services though
+	context.Lock()
+	if action, err := context.ApplicationACLs.GetMatchingAction(tcpPacket.DestinationAddress.To4(), tcpPacket.DestinationPort); err == nil && action&policy.Accept > 0 {
+		conn.SetState(TCPData)
+		return action, nil
+	}
+	context.Unlock()
+
+	// If we have an IP in the cache for the default match, the process here.
+	if _, err := context.externalIPCache.Get(tcpPacket.DestinationAddress.String()); err == nil {
```

Comment:
externalIPCache is accessed within context.Lock() (processNetworkSynAckPacket) and sometimes not (processApplicationSynPacket).